microcosm_flask/context.py

```diff
@@ -1,32 +1,22 @@
 from flask import request
-from microcosm.api import defaults
 
 
-X_REQUEST = "X-Request"
+# This is the total set of Globality-proprietary headers
+ALL_HEADER_LIST = (
+    "X-Request",  # These are values *invariant* to a user request, e.g. user-id, request-id, etc.
+    "X-Client",  # These are values *specific* to a single HTTP Client call to a web service
+)
+
+# This is the list of internal Globality Headers we use which are safe for logging.
+LOGGABLE_HEADER_WHITE_LIST = ("X-Request")
 
 
-def context_wrapper(include_header_prefix):
-    def retrieve_context():
+def make_get_request_context(header_white_list=LOGGABLE_HEADER_WHITE_LIST):
+    def get_request_context():
         return {
             header: value
+            for prefix in header_white_list
             for header, value in request.headers.items()
-            if header.startswith(include_header_prefix)
+            if header.startswith(prefix)
         }
-
-    return retrieve_context
-
-
-@defaults(
-    include_header_prefix=X_REQUEST,
-)
-def configure_request_context(graph):
-    """
-    Configure the flask context function which controls what data you want to associate
-    with your flask request context, e.g. headers, request body/response.
-
-    Usage:
-        graph.request_context()
-
-    """
-    include_header_prefix = graph.config.request_context.include_header_prefix
```

Comment:
Looks like we lose this config?

Reply:
Yes because we have never used it in over > 4 years and I think changing the header prefix requires thoughts about the design of the rest of the application. We could add it back, when we (or someone who depends on us) actually uses it.

Reply:
Basically what I found while doing this is the config we have is insufficient. I think if we wanted to make this available through the graph we'd want something smarter which supports getting the right type of context depending on various factors (e.g. for a given user flow, a given http request, a given handler execution).